script/install.sh

```diff
@@ -119,10 +121,39 @@ fetch() {
 
 # ------------------------------------------------------------------------------
 
+# Returns false if the script is invoked from a Windows command prompt.
+posix_terminal() {
+    # If run from a POSIX terminal (including under MSYS2 or Cygwin) the TERM
+    # variable is defined to something like "xterm". If run from a Windows
+    # command prompt through bash.exe from an MSYS installation, TERM keeps
+    # its default value, which is "cygwin".
+    if [[ $TERM == "cygwin" ]]; then
+        return 1;   # false
+    else
+        return 0;   # true
+    fi
+}
+
 COMMAND=
 COMPILER=dmd
 VERBOSITY=1
-ROOT=~/dlang
+if [ -z "$SCRIPT_PATH" ]; then
+    # We are being piped into bash from stdin. Set a default.
+    if posix_terminal; then
+        ROOT=~/dlang
+    else
+        # Default to a ROOT that is outside the POSIX-like environment.
+        ROOT=$(cygpath "$LOCALAPPDATA")/dlang
+    fi
+else
+    # Default to the script location.
+    if command -v cygpath &>/dev/null; then
```

Comment:
Isn't this done similarly in other places already? Maybe time to wrap in a function?